packages/extension/src/hosted/api/vscode/env/telemetry-ext.ts

```
@@ -71,24 +73,210 @@ export class TelemetryLogger {
+      this.logEvent(eventNameOrException, data);
+    } else {
+      this.sender.sendErrorData(eventNameOrException, data);
+    }
   }
 
   dispose(): void {
-    // TODO
+    if (this.sender?.flush) {
+      let tempSender: TelemetrySender | undefined = this.sender;
+      this.sender = undefined;
+      Promise.resolve(tempSender.flush!()).then((tempSender = undefined));
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)